src/Bem2d.jl

```diff
@@ -26,17 +26,81 @@ function obsgrid(xmin, ymin, xmax, ymax, npts)
     return xobs[:], yobs[:]
 end
 
+export Element
+struct Element
+    x1::Float64
+    y1::Float64
+    x2::Float64
+    y2::Float64
+    a::Float64
+    b::Float64
+    normalstress::Float64
+    # The following are derived properties (could be dynamic)
+    angle::Float64
+    length::Float64
+    halflength::Float64
+    xcenter::Float64
+    ycenter::Float64
+    rotmat::Array{Float64,2}
+    rotmatinv::Array{Float64,2}
+    xnormal::Float64
+    ynormal::Float64
+    xnodes::Array{Float64,1}
+    ynodes::Array{Float64,1}
+    # These are unused?
+    # uxconst::Float64
+    # uyconst::Float64
+    # uxquad::Array{Float64,1}
+    # uyquad::Array{Float64,1}
+end
+function Element(; x1, y1, x2, y2, a, b, normalstress)
+    dx = x2 - x1
+    dy = y2 - y1
+    magnitude = sqrt(dx^2 + dy^2)
+    angle = atan(dy, dx)
+    length = magnitude
+    halflength = 0.5 * length
+    xcenter = 0.5 * (x2 + x1)
+    ycenter = 0.5 * (y2 + y1)
+    rotmat = [cos(angle) -sin(angle) ; sin(angle) cos(angle)]
+    rotmatinv = [cos(-angle) -sin(-angle) ; sin(-angle) cos(-angle)]
+    xnormal = dy / magnitude
+    ynormal = -dx / magnitude
+    xnodes = [xcenter - (2 / 3 * dx / 2), xcenter, xcenter + (2 / 3 * dx / 2)]
+    ynodes = [ycenter - (2 / 3 * dy / 2), ycenter, ycenter + (2 / 3 * dy / 2)]
+    return Element(
+        x1,
+        y1,
+        x2,
+        y2,
+        a,
+        b,
+        normalstress,
+        angle,
+        length,
+        halflength,
+        xcenter,
+        ycenter,
+        rotmat,
+        rotmatinv,
+        xnormal,
+        ynormal,
+        xnodes,
+        ynodes,
+    )
+end
+
 export Elements
 mutable struct Elements
     x1::Array{Float64,1}
     y1::Array{Float64,1}
     x2::Array{Float64,1}
     y2::Array{Float64,1}
     name::Array{String,1}
-    uxconst::Array{Float64,1}
-    uyconst::Array{Float64,1}
-    uxquad::Array{Float64,2}
-    uyquad::Array{Float64,2}
+    uxconst::Array{Float64,1} # unused??
+    uyconst::Array{Float64,1} # unused??
+    uxquad::Array{Float64,2}  # unused??
+    uyquad::Array{Float64,2}  # unused??
```

Comment:
I found I didn't need these properties for the `src/ex_qdstepandsavetopo.jl` script.